gobblin-data-management/src/test/java/gobblin/data/management/copy/TimestampBasedCopyableDatasetTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.Lists;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.joda.time.DateTime;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import gobblin.data.management.copy.CopyConfiguration;
+import gobblin.data.management.copy.CopyableFile;
+import gobblin.data.management.policy.VersionSelectionPolicy;
+import gobblin.data.management.version.DatasetVersion;
+import gobblin.data.management.version.FileSystemDatasetVersion;
+import gobblin.data.management.version.TimestampedDatasetVersion;
+import gobblin.data.management.version.finder.DateTimeDatasetVersionFinder;
+import gobblin.data.management.version.finder.VersionFinder;
+import gobblin.dataset.Dataset;
+
+
+/**
+ * Test for {@link TimestampBasedCopyableDataset}.
+ */
+@Test(groups = { "gobblin.data.management.copy" })
+public class TimestampBasedCopyableDatasetTest {
+  @Test
+  public void testGetCopyableFiles() throws IOException {
+    Properties props = new Properties();
+    props.put(DateTimeDatasetVersionFinder.DATE_TIME_PATTERN_KEY, "yyyy/MM/dd");
+    props
+.put(TimestampBasedCopyableDataset.COPY_POLICY, TimeBasedCopyPolicyForTest.class.getName());
+    props.put(TimestampBasedCopyableDataset.DATASET_VERSION_FINDER,
+        TimestampedDatasetVersionFinderForTest.class.getName());
+    FileSystem localFs = FileSystem.getLocal(new Configuration());
+    TimestampBasedCopyableDataset copyabledataset =
+        new TimestampBasedCopyableDatasetForTest(localFs, props, new Path("/data/tracking/PVE"));
+    Collection<CopyableFile> copyableFiles = copyabledataset.getCopyableFiles(localFs, null);
+    Assert.assertEquals(copyableFiles.size(), 6);
+    Pattern pattern = Pattern.compile("dummy/[\\d]*/file[12]");
+    for (CopyableFile copyableFile : copyableFiles) {
```

Comment:
This check will return "true" if the getCopyableFiles() method returns the same file 6 times.